### PR TITLE
Refactor reviewer flags API into a single endpoint

### LINKED ISSUE #7419: Refactor reviewer flags APIs into a single endpoint
In https://github.com/mozilla/addons-server/issues/7282 I've added several API endpoints for reviewer flags. I've then starting adding more. @eviljeff [pointed out](https://github.com/mozilla/addons-server/pull/7407#issuecomment-361217316) it'd be more convenient to have a single reviewer flags endpoint with some parameters to change the values instead of adding a new endpoint every time, and he's absolutely right :)

### PR DESCRIPTION
**Pull request by diox**
Fix #7419

### CODE REVIEW

File: src/olympia/reviewers/templates/reviewers/review.html [hunk 1]
```diff
@@ -304,32 +304,40 @@ <h3 id="history">
                   id="force_enable_addon" type="button">{{ _('Force-enable add-on') }}</button>
         </li>
         <li {% if addon.auto_approval_disabled %}class="hidden"{% endif %}>
-          <button data-api-url="{{ url('reviewers-addon-disable-auto-approval', addon.pk) }}"
+          <button data-api-url="{{ url('reviewers-addon-flags', addon.pk) }}"
+                  data-api-method="patch"
+                  data-api-data="{&quot;auto_approval_disabled&quot;: true}"
```
Comment (eviljeff) [inline src/olympia/reviewers/templates/reviewers/review.html]: Hmm.  Is using `&quot;` better than `data-api-data='"auto_approval_disabled": true}'`?

Comment (diox) [reply src/olympia/reviewers/templates/reviewers/review.html]: I haven't tried it, I guess both works. I didn't want to use different quote characters than the rest of the HTML, do you have a strong opinion one way or the other?

Comment (eviljeff) [reply src/olympia/reviewers/templates/reviewers/review.html]: no strong opinions on this

Comment (diox) [reply src/olympia/reviewers/templates/reviewers/review.html]: I'll merge it as-is then as I've spent already way too much time on this refactor and don't want to retest everything...

**Pull request merged**